dbms/CMakeLists.txt

```diff
@@ -303,6 +303,7 @@ if (ENABLE_TESTS)
     )
     target_include_directories(gtests_dbms BEFORE PRIVATE ${SPARCEHASH_INCLUDE_DIR})
     target_compile_definitions(gtests_dbms PUBLIC DBMS_PUBLIC_GTEST)
+    target_compile_definitions(dbms PUBLIC DBMS_PUBLIC_GTEST)
```

Comment:
why add it?

Reply:
> why add it?

it is for join executor gtests.  In join executor spilling  gtests, the amount of data is small, so it is difficult to simulate  spilling, with the compilation option controlling that the spilling only goes to the second round and only half the partition spilling to disk.

Reply:
Isn't it controlled by `context.context.setSetting("max_bytes_before_external_join", Field(static_cast<UInt64>(max_bytes_before_external_join)));` at runtime?
Where is this compile option used?

Reply:
> Isn't it controlled by `context.context.setSetting("max_bytes_before_external_join", Field(static_cast<UInt64>(max_bytes_before_external_join)));` at runtime? Where is this compile option used?

Join.cpp  L1157 and L2576